lib/src/kbasesearchengine/parse/KeywordParser.java

```diff
@@ -33,6 +34,19 @@ public static ParsedObject extractKeywords(
             final ObjectLookupProvider lookup,
             final List<GUID> objectRefPath)
             throws IOException, ObjectParseException, IndexingException, InterruptedException {
+
+        // check pre-conditons
+        if(json == null) throw new IllegalArgumentException("json is a required parameter");
```

Comment:
You can use Utils.nonNull() for this. You might want to use Utils.notNullorEmpty() though.